src/main/java/seedu/address/logic/commands/EditPublicAddressCommand.java

```diff
@@ -0,0 +1,123 @@
+
+    private final Index index;
+    private final PublicAddress publicAddress;
+
+    /**
+     * Creates an EditPublicAddressCommand to edit the public address of a specified person.
+     *
+     * @param index         of the person in the filtered person list to edit
+     * @param network       network type of desired public address
+     * @param label         label of desired public address
+     * @param publicAddress new public address
+     */
+    public EditPublicAddressCommand(Index index, Network network, String publicAddress, String label) {
+        requireAllNonNull(index, network, label, publicAddress);
+
+        this.index = index;
+        this.publicAddress = PublicAddressFactory.createPublicAddress(network, publicAddress, label);
+    }
+
+    /**
+     * Executes the edit public address command by modifying the public address of the specified person.
+     *
+     * @param model Model where the command will be executed
+     * @return CommandResult indicating the result of the operation
+     * @throws CommandException if the command fails due to invalid input or state
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        Person personToEdit = getPersonToEdit(model);
+
+        if (!personToEdit.hasPublicAddressWithLabel(publicAddress.getNetwork(), publicAddress.label)) {
+            logger.warning(String.format(
+                    "Attempted to edit a public address with a non-matching label %1$s for network %2$s on person %3$s",
+                    publicAddress.label, publicAddress.getNetwork(), personToEdit.getName()));
+            throw new CommandException(
+                    String.format(MESSAGE_NON_MATCHING_LABEL, publicAddress.getNetwork(), personToEdit.getName()));
+        }
+
+        Person editedPerson = personToEdit.withUpdatedPublicAddress(publicAddress);
+        model.setPerson(personToEdit, editedPerson);
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(editedPerson)));
+    }
+
+    private Person getPersonToEdit(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        return lastShownList.get(index.getZeroBased());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditPublicAddressCommand)) {
+            return false;
+        }
+
+        EditPublicAddressCommand otherCommand = (EditPublicAddressCommand) other;
+        return index.equals(otherCommand.index) && publicAddress.equals(otherCommand.publicAddress);
+    }
+
+    @Override
```

Comment:
Can include network here since publicaddress might not differentiate between networks